<a href="https://colab.research.google.com/github/pitaconsumer/Health-pita/blob/master/CAPSTONE_COVID_19.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

**COVID-19 Recovery Influenced by East Mediterranean Nutritional Health**


import numpy as np
import pandas as pd
import scipy 
import matplotlib.pyplot as plt
%matplotlib inline

import seaborn as sns
from sqlalchemy import create_engine
import warnings
from pandas_profiling import ProfileReport

warnings.filterwarnings('ignore')
sns.set(style="whitegrid")

```
# This is formatted as code
```



In [44]:

!pip install pandas-profiling -U

Requirement already up-to-date: pandas-profiling in /usr/local/lib/python3.6/dist-packages (2.8.0)


In [0]:
import numpy as np
import pandas as pd
import scipy 
import matplotlib.pyplot as plt
%matplotlib inline

import seaborn as sns
from sqlalchemy import create_engine
import warnings
from pandas_profiling import ProfileReport

warnings.filterwarnings('ignore')
sns.set(style="whitegrid")



Although 86% of countries invest local or donor funds in health security, few countries pay for health security gap assessments and action plans out of national budgets. As such, we hypothesize that recovery rates are not only greatly influenced by the six dimensions of health security described in our analysis below, but also by the nutritional access and regional diets.  For example, the Eastern Mediterranean region consumes high level of nigella seeds, pomegrante seeds as well as almonds and pine nuts (categorized as 'tree nuts' in our Fat Supply Quantity data set), which are regionally believed to boost immunity. Perhaps, this boost in immunity could serve as another layer in fighting COVID-19 and result in higher recovery rates.  

# Upload Global Health Data Sets
## Need to upload the "Fat Supply Quantity" from Kaggle to add to 'Life Expectancy' data set.  
Source: https://www.kaggle.com/mariaren/covid19-healthy-diet-dataset/data https://storage.googleapis.com/kagglesdsdata/datasets/618335/1175348/Fat_Supply_Quantity_Data.csv?GoogleAccessId=web-data@kaggle-161607.iam.gserviceaccount.com&Expires=1590357735&Signature=bWQ%2Fj3yRSui127xJMXfJ%2BtW6L0L5xx68ngVOg8U3ewhDulVfTGGUSSVyTSQmO5%2FPnt0%2FAXI7zhGfc6Keejv7bDqaK5mB6pZX4YzqHWPfiBo4s8Oa20eVOXlr2fMFdAZSiOLa6d2fpcia9QfebPZflAH4SgGEO5nuqbJLj%2FDOF0Mv1VHkRRp2BZ8dWsogc2jqxDIqISkOnFxasYodcbzL0lKKzPcfF7QJy%2ByzZj3osBjQ0OXCLwjiAT2DdFdwe%2FQ7Bkthv8fNAW8w9uXExH79t%2BL4QV%2F%2F10BKldwlht3mvzg18ZQYIvrmMtA%2BVLsNYBcvSIH5p%2FJCGN1v6fKnTNZXsw%3D%3D&response-content-disposition=attachment%3B+filename%3DFat_Supply_Quantity_Data.csv

In [49]:
from google.colab import files 
uploaded = files.upload()

Saving Fat_Supply_Quantity_Data.csv to Fat_Supply_Quantity_Data (1).csv


In [0]:
import io 
df = pd.read_csv(io.BytesIO(uploaded['Fat_Supply_Quantity_Data.csv']))

In [0]:
df.head()


In [0]:
df.columns

In [0]:
postgres_user = 'dsbc_student'
postgres_pw = '7*.8G9QH21'
postgres_host = '142.93.121.174'
postgres_port = '5432'
postgres_db = 'lifeexpectancy'

engine = create_engine('postgresql://{}:{}@{}:{}/{}'.format(
    postgres_user, postgres_pw, postgres_host, postgres_port, postgres_db))

life_df = pd.read_sql_query('select * from lifeexpectancy',con=engine)

# no need for an open connection, 
# as we're only doing a single query
engine.dispose()


In [0]:
life_df

In [0]:
life_df.columns

In [0]:
#df.loc[df['shield'] > 6]
life_df.loc[life_df['Life expectancy '] > 70]

In [0]:
#ProfileReport(df)

In [0]:
#ProfileReport(life_df)

In [0]:
life_df2 = life_df['Year'].max()
life_df2

In [0]:
life_df.nunique()

In [0]:
df.nunique()

In [0]:
#Unique values
df['Recovered'].value_counts()

Exploratory Analysis 
Missing Data


In [52]:
#Drop less used from first data frame: Alcoholic Beverages, Sugar Crops, Sugar & Sweeteners ,Undernourished, Miscellaneous,Aquatic Products, Other,Undernourished,  
df2 = df.drop(['Alcoholic Beverages','Aquatic Products, Other','Sugar & Sweeteners','Sugar Crops','Miscellaneous','Undernourished','Aquatic Products, Other',], axis=1)
df2

Country  ...  Unit (all except Population)
0                           Afghanistan  ...                             %
1                               Albania  ...                             %
2                               Algeria  ...                             %
3                                Angola  ...                             %
4                   Antigua and Barbuda  ...                             %
..                                  ...  ...                           ...
165  Venezuela (Bolivarian Republic of)  ...                             %
166                             Vietnam  ...                             %
167                               Yemen  ...                             %
168                              Zambia  ...                             %
169                            Zimbabwe  ...                             %

[170 rows x 26 columns]

In [0]:
df2['Country']

We need to drop columns from dataframes that we have not imputed or filled in with values that produce "Nan". Missing data does not support Decision Tree models. 

In [53]:
#Drop less used from second data frame, Life data frame:
life_df2 = life_df.drop(['Measles ','Hepatitis B','infant deaths',' thinness  1-19 years','Alcohol', ' thinness 5-9 years','GDP','Total expenditure'], axis=1)
life_df2

Country  Year  ... Income composition of resources  Schooling
0     Afghanistan  2015  ...                           0.479       10.1
1     Afghanistan  2014  ...                           0.476       10.0
2     Afghanistan  2013  ...                           0.470        9.9
3     Afghanistan  2012  ...                           0.463        9.8
4     Afghanistan  2011  ...                           0.454        9.5
...           ...   ...  ...                             ...        ...
2933     Zimbabwe  2004  ...                           0.407        9.2
2934     Zimbabwe  2003  ...                           0.418        9.5
2935     Zimbabwe  2002  ...                           0.427       10.0
2936     Zimbabwe  2001  ...                           0.427        9.8
2937     Zimbabwe  2000  ...                           0.434        9.8

[2938 rows x 14 columns]

**Join datasets by key 'Country' **
We will join both dataframes and select only Year == 2015 from the life_df2 dataframe to create a new dataframe: covid_df.

In [0]:
covid_df = pd.merge(df2, life_df2[life_df2['Year'] == 2015], left_on = 'Country', right_on = 'Country')

We will run our newly merged dataset 'covid_df' that lists country records from 2015, our most recent year. We have 157 individual country records for the models.

In [0]:
covid_df

In [0]:
covid_df.corr()

*Analysis Note*:
We see that Highly correlated =or Above +/- .5 correlation includes 'Schooling' with 'animal fat'.

**Indicator of Health Index:** 
The availability of ventilators and access to critical care beds represents a huge factor towards a country's recovery rate in the COVID-19 pandemic. We found the Global Health Security Index, which is a composite of several healthcare dimensions that include: (1) Prevention; (2) Detection and Reporting; (3) Rapid Response; (4) Health System; (5) Compliance with International Norms; and (6) Risk Environment. A fuller explanation of these dimensions is seen in the note at the bottom. 

To measure the state of state and private health access as well as infrasturcture, we will use the 'Health Index' control for availability of ventilators, or other country financed health interventions and infrastructure to deal with health emergencies, like pandemics.


*Note: The study PREVENTION: Fewer than 7% of countries score in the
highest tier8 for the ability to prevent the emergence or
release of pathogens.
DETECTION AND REPORTING: Only 19% of countries
receive top marks for detection and reporting.
RAPID RESPONSE: Fewer than 5% of countries scored
in the highest tier for their ability to rapidly respond to
and mitigate the spread of an epidemic.
HEALTH SYSTEM: The average score for health system
indicators is 26.4 of 100, making it the lowest-scoring
category.
COMPLIANCE WITH INTERNATIONAL NORMS:
Less than half of countries have submitted
Confidence-Building Measures under the Biological
Weapons Convention (BWC) in the past three years,
an indication of their ability to adhere to important
international norms and commitments related to
biological threats.
RISK ENVIRONMENT: Only 23% of countries score
in the top tier for indicators related to their political
system and government effectiveness.  


~Source: Index: Global Healthy Security Index https://www.ghsindex.org/wp-content/uploads/2019/10/2019-Global-Health-Security-Index.pdf
    
~Variable: Ventilators

~Variable: Critical Care Beds
Saudi Arabia -22.8 per 100,000 (high)
Pakistan - 1.5 (lower-middle)
Iran - 4.6 (upper-middle medium)
Oman - 14.6 (high)
Yemen
Source: https://www.researchgate.net/figure/Number-of-critical-care-beds-per-100-000-population_fig1_338520008
expenditure health - https://link.springer.com/article/10.1007/s00134-012-2627-8

We will define a dictionary that includes the data from the Global Health Security Index report for the 195 countries reporting. Their respective scores will be represented and included as the column 'Health index' in the newly created data set: covid-df.

In [0]:

# Define a dictionary containing Health Index 
health_score_data = {'Country': ['United States','United Kingdom', 'Netherlands', 'Australia','Canada','Thailand','Sweden',
                                 'Denmark','South Korea','Finland','France','Slovenia','Switzerland','Germany','Spain','Norway',
                                 'Latvia','Malaysia','Belgium','Portugal','Japan', 'Brazil','Ireland','Singapore','Argentina',
                                 'Austria','Chile','Mexico','Estonia','Indonesia','Italy','Poland','Lithuania','South Africa',
                                 'Hungary','New Zealand','Greece','Croatia','Albania','Turkey','Serbia','Czech Republic','Georgia',
                                 'Armenia','Ecuador','Mongolia','Kyrgyz Republic','Saudi Arabia','Peru','Vietnam','China','Slovakia',
                                 'Philippines','Israel','Kenya','United Arab Emirates','India','Iceland','Kuwait','Romania',
                                 'Bulgaria','Costa Rica','Russia','Uganda','Colombia','El Salvador','Luxembourg','Montenegro','Morocco',
                                 'Panama','Liechtenstein','Myanmar','Laos','Lebanon','Nicaragua','Oman','Cyprus','Moldova',
                                 'Bosnia and Herzegovina','Jordan','Uruguay','Qatar','Kazakhstan','Ethiopia','Bhutan','Madagascar',
                                 'Egypt','Bahrain','Cambodia','North Macedonia','Dominican Republic','Sierra Leone','Zimbabwe','Ukraine',
                                 'Senegal','Nigeria','Iran','Malta','Trinidad and Tobago','Suriname','Tanzania','Bolivia',
                                 'Paraguay','Namibia',"Côte d'Ivoire",'Ghana','Pakistan','Belarus','St. Lucia','Cuba','Liberia','Nepal',
                                 'Bangladesh','Mauritius','Cameroon','Uzbekistan','Azerbaijan','Gambia','Rwanda','Sri Lanka','Maldives',
                                 'Tunisia','St. Vincent and The Grenadines','Micronesia','Guatemala','Guinea','Monaco','Brunei','Togo',
                                 'Afghanistan','Tajikistan','Niger','Barbados','Seychelles','Belize','Turkmenistan', 'Guyana','Haiti',
                                 'Botswana','San Marino','Swaziland','Bahamas','Andorra','Lesotho','Burkina Faso','Cabo Verde',
                                 'Antigua and Barbuda','Jamaica','Mali','Benin','Chad','Zambia','Mozambique','Malawi',
                                 'Papua New Guinea','Honduras','Grenada','Mauritania','Central African Republic','Comoros','Congo','Samoa',
                                 'St. Kitts and Nevis','Sudan','Vanuatu','Timor-Leste','Iraq','Fiji','Libya','Angola','Tonga',
                                 'Dominica','Algeria','Brazzaville','Djibouti','Venezuela','Burundi','Eritrea','Palau','South Sudan','Tuvalu',
                                 'Nauru','Solomon Islands','Niue','Cook Islands','Gabon','Guinea-Bissau','Syria','Kiribati',
                                 'Yemen','Marshall Islands','São Tomé and Príncipe','North Korea','Somalia','Equatorial Guinea'],
                     
        'Health_index': [83.5,77.9,75.6,75.5,75.3,73.2,72.1,70.4,70.2,68.7,68.2,67.2,67.0,66.0,65.9,64.6,62.9,62.2,61.0,
                        60.3,59.8,59.7,59.0,58.7,58.6,58.5,58.3,57.6,57.0,56.6,56.2,55.4,55.0,54.8,54.0,54.0,53.8,53.3,52.9,52.4,
                        52.3,52.0,52.0,50.2,50.1,49.5,49.3,49.3,49.2,49.1,48.2,47.9,47.6,47.3,47.1,46.7,46.5,46.3,46.1,45.8,45.6,
                        45.1,44.3,44.3,44.2,44.2,43.8,43.7,43.7,43.7,43.5,43.4,43.1,43.1,43.1,43.1,43.0,42.9,42.8,42.1,
                        41.3,41.2,40.7,40.6,40.3,40.1,39.9,39.4,39.2,39.1,38.3,38.2,38.2,38.0,37.9,37.8,37.7,37.3,36.6,36.5,36.4,
                        35.8,35.7,35.6,35.5,35.5,35.5,35.3,35.3,35.2,35.1,35.1,35.0,34.9,34.4,34.3,34.2,34.2,34.2,33.9,33.8,33.7,
                        33.0,32.8,32.7,32.7,32.7,32.6,32.5,32.3,32.3,32.2,31.9,31.9,31.8,31.8,31.7,31.5,31.1,31.1,31.1,30.6,30.5,
                        30.2,30.1,29.3,29.0,29.0,29.0,28.8,28.8,28.7,28.1,28.0,27.8,27.6,27.5,27.5,27.3,27.2,26.5,26.4,26.2,26.2,
                        26.1,26.0,25.8,25.7,25.7,25.2,25.1,24.0,23.6,23.6,23.2,23.0,22.8,22.4,21.9,21.7,21.6,20.8,20.7,20.5,
                        20.4,20.0,20.0,19.9,19.2,18.5,18.2,17.7,17.5,16.6,16.2]} 
  
# Convert the dictionary into DataFrame 
health_df = pd.DataFrame(health_score_data) 
  
# Using DataFrame.insert() to add the column into covid_df
#covid_df.insert(df2["health_score_data"]) 

#Merge the 2 data sets
merged_df = pd.merge(covid_df, health_df, left_on='Country', right_on='Country')

In [0]:
len(health_score_data['Country'])

In [0]:
merged_df.columns

In [0]:
df2['Recovered'].value_counts() 

**Missing Values**
Need to fill in missing values for Percentage Expenditure, BMI, Polio, Treenuts, Pulses, and Vegetables columns. 

In [0]:
life_df2['percentage expenditure'].fillna(life_df2['percentage expenditure'].mean(), inplace=True)

In [0]:
life_df2[' BMI '].fillna(life_df2[' BMI '].mean(), inplace=True)

In [0]:
life_df2['Polio'].fillna(life_df2['Polio'].mean(), inplace=True)

In [0]:
df2['Treenuts'].fillna(df2['Treenuts'].mean(),inplace=True)

In [0]:
df2['Pulses'].fillna(df2['Pulses'].mean(),inplace=True)

In [0]:
df2['Vegetables'].fillna(df2['Vegetables'].mean(),inplace=True)

In [0]:
df2['Meat'].fillna(df2['Meat'].mean(),inplace=True)

We will check the correlation between adjusted variables in new data set.

In [56]:
merged_df.corr()

Animal Products  ...  Health_index
Animal Products                         1.000000  ...      0.515481
Animal fats                             0.710333  ...      0.506550
Cereals - Excluding Beer               -0.478648  ...     -0.444253
Eggs                                    0.469318  ...      0.458177
Fish, Seafood                          -0.022442  ...     -0.068667
Fruits - Excluding Wine                -0.101656  ...     -0.104336
Meat                                    0.736385  ...      0.244104
Milk - Excluding Butter                 0.641814  ...      0.343129
Offals                                  0.029119  ...     -0.175541
Oilcrops                               -0.413549  ...     -0.382370
Pulses                                 -0.419936  ...     -0.267498
Spices                                 -0.209950  ...     -0.087904
Starchy Roots                          -0.387591  ...     -0.321883
Stimulants                              0.495903  ...      0.214624
Treenuts                                0.188401  ...      0.258659
Vegetal Products                       -1.000000  ...     -0.515501
Vegetable Oils                         -0.669238  ...     -0.127352
Vegetables                              0.071712  ...     -0.041928
Obesity                                 0.435854  ...      0.327343
Confirmed                               0.385509  ...      0.456909
Deaths                                  0.244084  ...      0.472358
Recovered                               0.348826  ...      0.355128
Active                                  0.249087  ...      0.347919
Population_x                            0.002143  ...      0.091874
Year                                         NaN  ...           NaN
Life expectancy                         0.634044  ...      0.650942
Adult Mortality                        -0.432882  ...     -0.413978
percentage expenditure                  0.118921  ...      0.051198
 BMI                                    0.437910  ...      0.367911
under-five deaths                      -0.199489  ...     -0.055520
Polio                                   0.398435  ...      0.369507
Diphtheria                              0.295216  ...      0.325217
 HIV/AIDS                              -0.375809  ...     -0.302289
Population_y                           -0.180596  ...      0.135824
Income composition of resources         0.701782  ...      0.696775
Schooling                               0.629505  ...      0.700796
Health_index                            0.515481  ...      1.000000

[37 rows x 37 columns]

In [57]:
merged_df.groupby(['Recovered','Vegetables','Treenuts']).mean

<bound method GroupBy.mean of <pandas.core.groupby.generic.DataFrameGroupBy object at 0x7f050ce03da0>>

Decision Tree Regression Model

In [58]:
from sklearn import tree
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeRegressor
from sklearn.metrics import accuracy_score 

#X = merged_df[['percentage expenditure',' BMI ','Polio','Treenuts','Pulses','Meat','Vegetables','Health_index']]
#X = merged_df['Treenuts','Health_index'] #Dropping a bunch to check
merged_df_dropped = merged_df.dropna()
# X = merged_df_dropped[['Treenuts', 'Health_index']]
X = merged_df_dropped.drop(columns=['Recovered', 'Country', 'Unit (all except Population)', 'Status', 'Confirmed', 'Population_x', 'Population_y'])
y = merged_df_dropped['Recovered']

#Split data to train and test with 20% sample 
X_train, X_test, y_train, y_test = train_test_split(X,y, test_size=.2,random_state =5)

#regressor = DecisionTreeRegressor()
dtr_model = DecisionTreeRegressor(random_state=0)
dtr_model.fit(X_train, y_train)  

DecisionTreeRegressor(ccp_alpha=0.0, criterion='mse', max_depth=None,
                      max_features=None, max_leaf_nodes=None,
                      min_impurity_decrease=0.0, min_impurity_split=None,
                      min_samples_leaf=1, min_samples_split=2,
                      min_weight_fraction_leaf=0.0, presort='deprecated',
                      random_state=0, splitter='best')

In [0]:
(merged_df_dropped['Recovered'] > 0.5)

Run our score on training set of Decision Tree Regression model.

In [59]:
print(dtr_model.score(X_train, y_train))

0.9999981172875818


Run our score on on test set of Decision Tree Regression model.

In [60]:
print(dtr_model.score(X_test, y_test))

0.10989889981167167


We will identify the most important features in the training set in our Decision Tree Regression model. We see that our dependent variable of interest "Tree Nuts" did not rank in the top 10. This means it does not have as much significance as we hypothesized.

In [61]:
pd.Series(dtr_model.feature_importances_, index=X_train.columns).sort_values(ascending=False)

Life expectancy                    0.403328
Pulses                             0.340356
Cereals - Excluding Beer           0.101938
Deaths                             0.080202
Fruits - Excluding Wine            0.017273
Animal fats                        0.013204
Eggs                               0.009714
Stimulants                         0.008631
Milk - Excluding Butter            0.007371
Vegetables                         0.007050
Fish, Seafood                      0.003941
Schooling                          0.001822
Meat                               0.000952
Income composition of resources    0.000938
Obesity                            0.000899
under-five deaths                  0.000726
Polio                              0.000454
Active                             0.000344
Starchy Roots                      0.000193
 BMI                               0.000182
Offals                             0.000128
Vegetable Oils                     0.000112
Adult Mortality                 

Run Second Model: Linear Regression Model
We will use the same target variable 'Recovered' and the above dependent variables. 

In [62]:
from sklearn.linear_model import LinearRegression

line = LinearRegression()

line.fit(X_train, y_train)

LinearRegression(copy_X=True, fit_intercept=True, n_jobs=None, normalize=False)

We will check the score of our Linear Regression model on our same training data set (X_train). 

In [63]:
line.score(X_train, y_train)

0.5045183791666261

We will check the score of our Linear Regression model on our test training set. We see that the score is much lower on our test set. This means that our second model overfit the data to the training set.

In [64]:
line.score(X_test, y_test)

0.14942329478116556

We will set a thresshold of 'Recovery' to 10 percent to see countries that fall into that classifying group.

In [65]:
merged_df[merged_df['Recovered'] <= 0.1]

Country  Animal Products  ...  Schooling  Health_index
0            Afghanistan          21.6397  ...       10.1          32.3
1                Albania          32.0002  ...       14.2          52.9
2                Algeria          14.4175  ...       14.4          23.6
3                 Angola          15.3041  ...       11.4          25.2
4    Antigua and Barbuda          27.7033  ...       13.9          29.0
..                   ...              ...  ...        ...           ...
139              Uruguay          25.5069  ...       15.5          41.3
140           Uzbekistan          25.9903  ...       12.1          34.3
142                Yemen          12.5401  ...        9.0          18.5
143               Zambia           9.6005  ...       12.5          28.7
144             Zimbabwe          10.3796  ...       10.3          38.2

[119 rows x 40 columns]

**Sorting Eastern Mediterranean Countries**
We are curious about a region with a high 'Treenuts' content in their daily diet. So we selected the Eastern Mediterranean region from the World Health Organization grouping.

In [0]:
merged_df.iloc[[0,40,70,71,77,83,87,103,116,117,133],'Treenuts'].mean()

In [79]:
merged_df[merged_df['Country'].isin(['Lebanon', 'Afghanistan','Kuwait', 'Pakistan', 'Saudi Arabia','Jordan','Syria','Yemen','Egypt','United Arab Emirates','Oman','Bahrain','Qatar','Morocco','Libya','Tunisia','Somalia','Iran','Iraq'])]

Country  Animal Products  ...  Schooling  Health_index
0             Afghanistan          21.6397  ...       10.1          32.3
40                  Egypt          15.9979  ...       13.1          39.9
64                   Iraq          11.1327  ...       10.1          25.8
70                 Jordan          12.2622  ...       13.1          42.1
74                 Kuwait          19.8923  ...       13.3          46.1
76                Lebanon          12.4024  ...       13.3          43.1
92                Morocco          15.4227  ...       12.1          43.7
103                  Oman          23.6146  ...       13.7          43.1
104              Pakistan          24.8245  ...        8.1          35.5
114          Saudi Arabia          15.3954  ...       16.1          49.3
133               Tunisia          12.7886  ...       14.6          33.7
138  United Arab Emirates          21.4437  ...       13.3          46.7
142                 Yemen          12.5401  ...        9.0          18.5

[13 rows x 40 columns]

In [80]:
merged_df.loc[[0,40,64,70,74,76,92,103,104,114,133,138,142],'Treenuts']

0      0.7513
40     0.2992
64     0.6341
70     0.9018
74     1.1810
76     2.9110
92     1.9048
103    0.6107
104    0.1909
114    0.7128
133    2.3289
138    3.8246
142    0.0769
Name: Treenuts, dtype: float64

We see that the Eastern Mediterranean countries included (Afghanistan, Egypt, Iraq, Jordan, Kuwait, Lebanon, Morocco, Oman, Pakistan, Saudi Arabia, Tunisia, United Arab Emirates, and Yemen) present a range between .076 (Yemen) to 3.82 (UAE) regarding 'Treenuts' composite in diet. This range may be attributed to income per capita. Specifically, the lowest end of spectrum is Yemen, a lower-income country, in contrast to the highest end of spectrum UAE, a high-income country. We would need to do a comparative analysis to see if the same trend occurs with this diet in another region and see if income disproportionally affects the consumption of tree nuts. 